src/reference/00-Getting-Started/07A-Scopes.md

```diff
@@ -393,3 +375,178 @@ lazy val util = (project in file("util"))
+    },
+    scalacOptions in console -= "-Ywarn-unused-import",
+    scalacOptions in Compile := scalacOptions.value // added by sbt
+  )
+```
+
+The answer is `List(-Ywarn-unused-import)`. Because of Rule 1B `scalacOptions in Compile`
+has higher precedence than `scalacOptions in console`.
+
+- Rule 2: For Dependency configuration axis, sbt searches in the order of the given value,
+  its parents, their parents recursively, and then falls back to `Global`.
+
+The example for that is `projA` that we saw earlier:
+
+```scala
+lazy val foo = settingKey[Int]("")
+lazy val bar = settingKey[Int]("")
+
+lazy val projA = (project in file("a"))
+  .settings(
+    foo := {
+      (bar in Test).value + 1
+    },
+    bar in Compile := 1
+  )
+```
+
+`(bar in Test)` is undefined, but due to Rule 2 it will look for `(bar in Runtime)` first,
+which is `Test` configuration's parent, then `(bar in Compile)`, which is its parent.
+
+- Rule 3: For Task axis, sbt searches for the given value, and then falls back to `Global`,
+  which is non-task scoped version of the scope.
+
+We can apply Rule 3 to think about the `(scalacOptions in (Compile, console))` example.
+Since `(scalacOptions in (Compile, console))` was not there, Rule 3 says that one of
+the delegates is `(scalacOptions in Compile)`.
+
+These are well defined rules, but you might want to look up quickly what is going on.
+This is where `inspect` can be used.
+
+```
+Hello> inspect projC/compile:console::scalacOptions
+[info] Task: scala.collection.Seq[java.lang.String]
+[info] Description:
+[info]  Options for the Scala compiler.
+[info] Provided by:
+[info]  {file:/Users/xxxx/}projC/compile:scalacOptions
+[info] Defined at:
+[info]  /Users/xxxx/build.sbt:47
+[info] Reverse dependencies:
+[info]  projC/compile:console
+[info]  projC/*:test
+[info] Delegates:
+[info]  projC/compile:console::scalacOptions
+[info]  projC/compile:scalacOptions
+[info]  projC/*:console::scalacOptions
+[info]  projC/*:scalacOptions
+[info]  {.}/compile:console::scalacOptions
+[info]  {.}/compile:scalacOptions
+[info]  {.}/*:console::scalacOptions
+[info]  {.}/*:scalacOptions
+[info]  */compile:console::scalacOptions
+[info]  */compile:scalacOptions
+[info]  */*:console::scalacOptions
+[info]  */*:scalacOptions
+....
+```
+
+See "Provided by". This shows that `projC/compile:console::scalacOptions` is
+actually delegated to `projC/compile:scalacOptions`.
+Next see the "Delegates". This is *all* of the possible delegate candidates
+listed in the order of precedence.
+
+- All the scopes with `projC` scoping on Subproject axis are listed first,
+  then `ThisBuild` (`{.}`), and `Global` (`*`).
+- Within a subproject, scopes with `Compile` scoping on Dependency configuration axis
+  are listed first, then falls back to `Global` (`*`) configuration.
+- Finally, Task axis scoping lists the given task value `console::` and the one without.
+
+Note that scope delegation feels similar to class inheritance in an object-oriented language,
```

Comment:
I suggest two things:
  
1. Make this a top-level section.
2. Avoid the term "dynamic dispatch" and describe what it really means. The example given is great, but it's difficult to parse for people because it involves understanding what dynamic dispatch. I would assume lots of people don't know what it is.
3. Write a sbt example that shows this behaviour.